### DISCUSSION TITLE
degrees are converted to radians with a trivial transformation - why is this correct?

### DISCUSSION THREAD

**Discussion by storm-ptr** [boostorg/geometry#1328] [Q&A] [answered]
code
```
#include <boost/geometry.hpp>
#include <boost/geometry/srs/epsg.hpp>
#include <boost/geometry/srs/projection.hpp>
#include <iostream>

using point_t = boost::geometry::model::d2::
    point_xy<double, boost::geometry::cs::geographic<boost::geometry::degree>>;

using proj_t =
    boost::geometry::srs::projection<boost::geometry::srs::static_epsg<4326>>;

int main()
{
    auto pj = proj_t{};
    auto a = point_t{-123., 20.};
    auto b = point_t{};
    pj.forward(a, b);
    std::cout << "{.lon=" << b.x() << ", .lat=" << b.y() << "}\n";
}
```

output
```
{.lon=-2.14675, .lat=0.349066}
```

[godbolt](https://godbolt.org/z/x9on1h9W8)

**Comment by vissarion** [comment, marked as answer]
This function should not be used this way. When projecting you want to transform a geometry from a spheroid or a projected coordinate system to another projected coordinate system. That is the result should always be a two dimensional point (in cartesian coordinates and in units defined by the target coordinate system). One such a system is the Web Mercator projection (epsg: 3857). So by changing your example a bit 
```cpp
#include <boost/geometry.hpp>
#include <boost/geometry/srs/epsg.hpp>
#include <boost/geometry/srs/projection.hpp>
#include <iostream>

using geographic_point_t = boost::geometry::model::d2::
    point_xy<double, boost::geometry::cs::geographic<boost::geometry::degree>>;

using cartesian_point_t = boost::geometry::model::d2::
    point_xy<double, boost::geometry::cs::cartesian>;

//web mercator projection
using proj_t =
    boost::geometry::srs::projection<boost::geometry::srs::static_epsg<3857>>;

int main()
{
    auto pj = proj_t{};
    auto a = geographic_point_t{-123., 20.};
    auto b = cartesian_point_t{};
    pj.forward(a, b);
    std::cout << b.x() << " , " << b.y() << "\n";
}
```
output
```
-1.01328e+07, 9.8602e+06
```
which is as expected. The projection source is assumed to be WGS84 coordinate system (epsg: 4326).

**Comment by storm-ptr** [comment]
I agree that my example is far-fetched. I accidentally stumbled upon this myself, in tests. It's just that nowhere in the output parameters, neither in the projection nor in the geometry, are radians indicated, only degrees.

**Comment by awulkiew** [comment]
When cartesian points are used in spherical or geographic context the angle unit is radian. So if a result of transformation are longitude and latitude and you pass cartesian point instead of geographic one then unit is converted if necessary.

**Comment by storm-ptr** [comment]
You are right - the unit of measurement for the angle of cartesian points is the radian.
But I used the geographic type of point with degrees.

```
// cartesian or undefined
template <typename CoordinateSystem>
struct cs_angular_units
{
    typedef geometry::radian type;
};

template<typename DegreeOrRadian>
struct cs_angular_units<cs::geographic<DegreeOrRadian> >
{
    typedef DegreeOrRadian type;
};
```

[boost.org](https://www.boost.org/doc/libs/1_86_0/boost/geometry/core/cs.hpp)

Here is another example of a trivial transformation, but one that preserves degrees.

code
```
#include <boost/geometry.hpp>
#include <boost/geometry/srs/epsg.hpp>
#include <boost/geometry/srs/transformation.hpp>
#include <iostream>

using point_t = boost::geometry::model::d2::
    point_xy<double, boost::geometry::cs::geographic<boost::geometry::degree>>;

using proj_t = boost::geometry::srs::transformation<
    boost::geometry::srs::static_epsg<4326>,
    boost::geometry::srs::static_epsg<4326>>;

int main()
{
    auto pj = proj_t{};
    auto a = point_t{-123., 20.};
    auto b = point_t{};
    pj.forward(a, b);
    std::cout << "{.lon=" << b.x() << ", .lat=" << b.y() << "}\n";
}
```

output
```
{.lon=-123, .lat=20}
```

[godbolt](https://godbolt.org/z/sfTWoEP7j)

**Comment by awulkiew** [comment]
Ah right. I looked at Vissarion's example. So my answer is incorrect. Sorry for confusion.

Vissarion is right, i.e. forward projection always projects to cartesian and sets the result directly with `bg::set`:
https://github.com/boostorg/geometry/blob/develop/include/boost/geometry/srs/projections/impl/pj_fwd.hpp#L97-L101
